Detectors/TOF/calibration/testWorkflow/LHCClockCalibratorSpec.h

```diff
@@ -71,28 +71,23 @@ class LHCClockCalibDevice : public o2::framework::Task
 
   void run(o2::framework::ProcessingContext& pc) final
   {
-    static const double TFlengthInv = 1E6 / o2::raw::HBFUtils::Instance().getNOrbitsPerTF() / o2::constants::lhc::LHCOrbitMUS;
-
-    auto tfcounter = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("input").header)->startTime;
     auto data = pc.inputs().get<gsl::span<o2::dataformats::CalibInfoTOF>>("input");
+    o2::base::TFIDInfoHelper::fillTFIDInfo(pc, mCalibrator->getCurrentTFInfo());
 
     if (!mcalibTOFapi) {
       mcalibTOFapi = new o2::tof::CalibTOFapi(long(0), &mPhase, &mTimeSlewing); // TODO: should we replace long(0) with tfcounter defined at the beginning of the method? we need the timestamp of the TF
       mCalibrator->setCalibTOFapi(mcalibTOFapi);
     }
-
     if (data.size() == 0) {
       return;
     }
-
-    tfcounter = uint64_t(data[0].getTimestamp() * TFlengthInv);
```

Comment:
@chiarazampolli @noferini : what is the meaning of this line? The timestamp you extract is integer in seconds, how you can convert it to "tfcounter" and why?
Also, we should not use o2::raw::HBFUtils in the production workflows, it is not guaranteed to have correct values there.